push/topology.go

```diff
@@ -69,20 +68,14 @@ func PushTopology(c echo.Context) error {
 		return dutyAPI.WriteError(c, dutyAPI.Errorf(dutyAPI.EINTERNAL, "error saving components: %v", dutydb.ErrorDetails(err)))
 	}
 
-	var idsToDelete []string
-	if err := ctx.DB().Model(&models.Component{}).Select("id").Where("topology_id = ?", data.TopologyID).Where("deleted_at IS NULL").Where("id NOT IN ?", returnedIDs).Find(&idsToDelete).Error; err != nil {
+	// Soft delete component ids which were not created
+	if err := ctx.DB().Model(&models.Component{}).
```

Comment:
@moshloop Should we just hard delete these components ? Or let cleanup do its thing ?

Reply:
I think its better to let cleanup handle it

Reply:
👍🏽 then lets merge it